image-processing/UI.py

```diff
@@ -1,484 +1,202 @@
-                        continue
-
-                    break
-
-                while True:
-                    set_parameters = input("Would you like to use preset values or set your own parameters? ('yes' for presets, 'no' for custom parameters) ").lower()
-
-                    if not set_parameters in ('yes', 'no'):
-                        yesno_error()
-                        continue
-
-                    break
-
-
-                for scan in range(len(scans_saved)):
-
-                    if set_parameters == 'yes':
+    def input_sample(self):
+        sample_type = input("Type of Sample: ")
+        
+        if sample_type in [str(n) for n in range(len(self.material_types) + 1)]:
+            print("Please ensure samples are scanned at 75dpi, or the next closest possible resolution.\n")
+            return int(sample_type)
 
-                            original = scans_saved[scan][0]
-                            sigma = 1
-                            row = 250
-                            finder = linefinder(original, sigma, row)
+        # elif sample_type.lower() == "back":
+        #     self.get_path()
+            
+        else:
+            print(f"Ensure an integer between 1 and {len(self.material_types)} inclusive is entered, please try again.\n")
+            self.input_sample()
+    
+    def input_grouping(self):
+        grouping = input(f"Does the folder contain scans of low or high areal weight sample? Enter 'low' or 'high': ").lower()
+        if grouping in ('high', 'low'):
+            return grouping
+        
+        # elif grouping.lower() == "back":
+        #     self.input_sample()
 
-                            if view_plot == 'yes':
-                                print('Result for Sample: {}'.format(paths_saved[scan].stem))
+        else:
+            print("Response not recognised, please respond with either 'low' or 'high'.\n")
+            self.input_grouping()
+    
+    def input_view_plot(self):
+        view_plot = input("Would you like to view the output plot from the analysis software? ").lower()
 
-                                if sampleType == 1:
-                                    finder.severity_carbon(4, scans_saved[scan][1])
+        if view_plot in ('yes', 'no'):
+            return view_plot
+        
+        # elif view_plot.lower() == "back":
+        #     self.input_grouping()
 
-                                if sampleType == 2:
-                                    finder.severity_metal_coated(6, scans_saved[scan][1])
+        else:
+            self.yesno_error()
+            self.input_view_plot()
 
-                                name = str(paths_saved[scan].stem)
-                                finder.plot_nice(name = name)
+    def input_baseline(self):
+        baseline = input('\nWhat severity out of 10 is required to fail a sample? \nRecommendations: 4 for Carbon Veil, 6 for Metal-Coated Carbon Veil \nEnter Value: ')    #got rid of divide by 2 for now
```

Comment:
These don't really make sense as "out of 10" recommendations - this is the exact value the mean prominence is compared to. Doesn't fit as an out of 10 recommendation as it won't match the out of 10 score returned :)

Reply:
ahh gotcha, thanks for the clarification! sorted (TF)